V5.27 documentation (#3899)
mattermost/docs: source/administration/changelog.md

Also see [changelog in progress](http://bit.ly/2nK3cVf) for the next release.

## Release v5.27 - [Quality Release](https://docs.mattermost.com/process/release-faq.html#release-overview)

**Release day: 2020-09-16**

Mattermost v5.27.0 contains a low level security fix. [Upgrading](http://docs.mattermost.com/administration/upgrade.html) is recommended. Details will be posted on our [security updates page](https://about.mattermost.com/security-updates/) 30 days after release as per the [Mattermost Responsible Disclosure Policy](https://www.mattermost.org/responsible-disclosure-policy/).

### Improvements

>  - Added the ability to upgrade Mattermost from Team Edition to Enterprise Edition E0 directly from the System Console.

Reviewer: ```suggestion
 - Added the ability to upgrade Mattermost from Team Edition to Enterprise Edition directly from the System Console.
```

>  - Added various improvements for Admin Advisor feature (Team Edition), including that the bot messages now appear only once for the 500-user advisory and the banner nagging interval is reduced from daily to weekly.

Reviewer: ```suggestion
 - Added various improvements for Admin Advisor feature (Team Edition), including that the bot messages now appear only once for the 500-user advisory and the banner notification interval is reduced from daily to weekly.
```

 - Changed the Default Theme setting in the System Console to a drop-down field.

### Bug Fixes
 - Fixed an issue where the server crashed when a Compliance Export job was run for Global Relay EML.

>  - Fixed an issue where Compliance Jobs did not restart correctly after a Warning status.

Reviewer: ```suggestion
 - Fixed an issue where Compliance Jobs did not restart correctly after a `Warning`status.
```

 - Fixed an issue where users were not matching on mixed-case SAML assertions.
 - Fixed an issue where Channel Admin was not able to make the default role as Channel Admin for AD/LDAP Groups.

>  - Fixed an issue where user's role was not added correctly in the Members block in **System Console > Teams** System Console page.

Reviewer: ```suggestion
 - Fixed an issue where user role was not added correctly in the Members block in **System Console > Teams**.
```

>  - Fixed an issue where a team stopped loading in the System Console "Filter By"-dropdown when a search was performed and then cleared.

Reviewer: ```suggestion
 - Fixed an issue where a team stopped loading in the System Console **Filter By**-dropdown when a search was performed and then cleared.
```

>  - Fixed an issue where the ability to demote Admins to members and to deactivate accounts from **System Console > Users** were not available.

Reviewer: ```suggestion
 - Fixed an issue where the ability to demote Admins to members and to deactivate accounts from **System Console > Users** was not available.
```

 - Fixed an issue where a false message "Group Mentions is already taken" was shown when a System Admin tried to add a channel to an AD/LDAP Group.
 - Fixed an issue where a AD/LDAP group mention of an outsider group was highlighted in a Group Synced channel.
 - Fixed an issue where incoming webhooks owned by a bot did not consistently allow a username override.
 - Fixed an issue where the emoji picker in the Edit Post modal was misaligned.
 - Fixed an issue where pasted unicode emojis failed to appear once posted.
 - Fixed an issue where long text in message edit modal did not scroll with a scroll bar.
 - Fixed an issue with Accessibility where user's name was not displayed in alt text on some images.
 - Fixed an issue where dates on **System Console > Site Statistics - Dates** were displayed out of order on days when there were no posts.
 - Fixed an issue where the Admin Advisor bot was unexpectedly displayed in the **Integrations > Bot Accounts** page.
 - Fixed an issue where a new badge in the channel sidebar category header reappeard after a channel was removed from the category.

>  - Fixed an issue where the theme color for "Sidebar Text Active Border" was not currently being used in the active border in the sidebar.

Reviewer: ```suggestion
 - Fixed an issue where the theme color for **Sidebar Text Active Border** was not currently being used in the active border in the sidebar.
```

 - Fixed an issue where users saw an incorrect mention count when added to a channel by another user.
 - Fixed an issue where channels created from another browser tab did not immediately appear in the channel sidebar.
 - Fixed an issue where a console error showed when creating a new custom category in the channel sidebar.
 - Fixed an issue where enabling the new channel sidebar created invalid channel links.
 - Fixed an issue where a channel state got broken after an "unallowed" deletion.
 - Fixed an issue where dynamic slash command autocomplete options did not update between requests.

>  - Fixed an issue where an incorrect callback URL with OAuth 2.0 allowed users to click "Back to Mattermost" in the authentication window.

Reviewer: ```suggestion
 - Fixed an issue where an incorrect callback URL with OAuth 2.0 allowed users to click **Back to Mattermost** in the authentication window.
```

 - Fixed an issue where a panic was caused when a user joined a team with default channels archived.
 - Fixed an issue where ``App.GetSidebarCategories()`` panicked on nil returned value.
 - Fixed an issue where the ``SendEmailNotifications`` setting blocked testing the SMTP connection.

### Open Source Components
 - Removed ``@types/redux-mock-store`` and ``tinycolor2`` from https://github.com/mattermost/mattermost-webapp.
 - Added ``bootstrap-colorpicker`` in https://github.com/mattermost/mattermost-webapp.
 - Added ``@react-native-community/clipboard`` in https://github.com/mattermost/mattermost-mobile.

### API Changes
 - Added ``POST api/v4/upgrade_to_enterprise`` API endpoint to be able to execute an inplace upgrade from Team Edition to Enterprise Edition.
 - Added ``GET api/v4/upgrade_to_enterprise/status`` API endpoint to get the current status for the inplace upgrade from Team Edition to Enterprise Edition.
 - Added ``POST api/v4/restart`` API endpoint to restart the system after an upgrade from Team Edition to Enterprise Edition.
 
### Known Issues
 - Twitter link previews do not work in Mattermost.

>  - On a server using a subpath, the URL opens a blank page if the System Admin changes the Site URL in the System Console UI. To fix, the System Admin should restart the server.

Reviewer: ```suggestion
 - On a server using a subpath, the URL opens a blank page if the System Admin changes the Site URL in the System Console. To fix this, the System Admin should restart the server.
```

 - Login does not work when Custom Terms of Service is enabled and MFA is enforced.
 - Google login fails on the Classic mobile apps.

>  - Status may sometimes get stuck as away or offline in High Availability mode with IP Hash turned off.

Reviewer: ```suggestion
 - Status may sometimes get stuck as Away or Offline in High Availability mode with IP Hash turned off.
```
